stdlib/std.jsonnet

```diff
@@ -864,19 +889,61 @@ limitations under the License.
         local lines = ['{\n']
                       + std.join([',\n'],
                                  [
-                                   [cindent + indent + '"' + k + '": '
+                                   [cindent + indent + std.escapeStringJson(k) + ': '
                                     + aux(v[k], path + [k], cindent + indent)]
                                    for k in std.objectFields(v)
                                  ])
                       + ['\n' + cindent + '}'];
         std.join('', lines);
     aux(value, [], ''),
 
+  manifestYamlDoc(value)::
+    local aux(v, in_array, in_object, path, cindent) =
+      if v == true then
+        'true'
+      else if v == false then
+        'false'
+      else if v == null then
+        'null'
+      else if std.type(v) == 'number' then
+        '' + v
+      else if std.type(v) == 'string' then
+        local len = std.length(v);
+        if len == 0 then
+          '""'
+        else if v[len - 1] == '\n' then
+          local split = std.split(v, '\n');
+          std.join('\n' + cindent, ['|'] + split[0:std.length(split) - 1])
+        else
+          std.escapeStringJson(v)
+      else if std.type(v) == 'function' then
+        error 'Tried to manifest function at ' + path
```

Comment:
👍

Reply:
If we had try / catch we would add the context into those errors too